src/main/java/org/folio/service/pieces/PiecesClaimingService.java

```diff
@@ -0,0 +1,237 @@
+    return StreamEx.of(piecesByVendorList).distinct().filter(Objects::nonNull)
+      .groupingBy(Pair::getKey, mapping(Pair::getValue, toList()));
+  }
+
+  private Future<ClaimingResults> createJobsByVendor(JsonObject config, Map<String, List<String>> pieceIdsByVendorId,
+                                                     RequestContext requestContext) {
+    log.info("createJobsByVendor:: Creating jobs by vendor, vendors by pieces count: {}", pieceIdsByVendorId.size());
+    if (CollectionUtils.isEmpty(pieceIdsByVendorId)) {
+      log.info("createJobsByVendor:: No jobs are created, pieceIdsByVendorId is empty");
+      return Future.succeededFuture(new ClaimingResults()
+        .withClaimingPieceResults(createErrorClaimingResults(pieceIdsByVendorId, CANNOT_GROUP_PIECES_BY_VENDOR_MESSAGE.getValue())));
+    }
+    return collectResultsOnSuccess(createUpdatePiecesAndJobFutures(config, pieceIdsByVendorId, requestContext))
+      .map(updatedPieceLists -> {
+        if (CollectionUtils.isEmpty(updatedPieceLists)) {
+          log.info("createJobsByVendor:: No pieces were processed for claiming");
+          return new ClaimingResults().withClaimingPieceResults(createErrorClaimingResults(pieceIdsByVendorId, CANNOT_CREATE_JOBS_AND_UPDATE_PIECES.getValue()));
+        }
+        var successClaimingPieceResults = createSuccessClaimingResults(updatedPieceLists);
+        log.info("createJobsByVendor:: Successfully processed pieces for claiming, count: {}", successClaimingPieceResults.size());
+        var claimingResults = new ClaimingResults().withClaimingPieceResults(successClaimingPieceResults);
+        log.info("createJobsByVendor:: Returning claiming results, claimingResults: {}", JsonObject.mapFrom(claimingResults).encodePrettily());
+        return claimingResults;
+      });
+  }
+
+  private List<Future<List<String>>> createUpdatePiecesAndJobFutures(JsonObject config, Map<String, List<String>> pieceIdsByVendorId,
+                                                                     RequestContext requestContext) {
+    var updatePiecesAndJobFutures = new ArrayList<Future<List<String>>>();
+    pieceIdsByVendorId.forEach((vendorId, pieceIds) -> config.stream()
+      .filter(entry -> isExportTypeClaimsAndCorrectVendorId(vendorId, entry) && Objects.nonNull(entry.getValue()))
+      .forEach(entry -> {
+        log.info("createJobsByVendor:: Preparing job integration detail for vendor, vendor id: {}, pieces: {}, job key: {}",
+          vendorId, pieceIds.size(), entry.getKey());
+        updatePiecesAndJobFutures.add(updatePiecesAndCreateJob(pieceIds, entry, requestContext));
+      }));
+    return updatePiecesAndJobFutures;
+  }
+
+  private static ClaimingResults createEmptyClaimingResults(String message) {
+    return new ClaimingResults().withClaimingPieceResults(List.of(new ClaimingPieceResult().withError(new Error().withMessage(message))));
```

Comment:
maybe name meethod to createEmptyErrorClaimingResults or createErrorClaimingResults or other to indicate we are creating error

Reply:
> maybe name meethod to createEmptyErrorClaimingResults or createErrorClaimingResults or other to indicate we are creating error

There is already another method with name `createErrorClaimingResults`, this one have bit different role